scripts/release/controller/setup.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+if [ -z "${BUILDTIMESTAMP}" ]; then
+    date "+%Y%m%d_%H%M%S" > "${HOME}/buildtimestamp"
+    BUILDTIMESTAMP=$(cat "${HOME}/buildtimestamp")
+    export BUILDTIMESTAMP
+    echo run "${0}" with output to "${HOME}/buildlog_${BUILDTIMESTAMP}"
+    (bash "${0}" "${1}" "${2}" 2>&1) | tee "${HOME}/buildlog_${BUILDTIMESTAMP}"
+    exit 0
+fi
+
+date "+setup start %Y%m%d_%H%M%S"
+
+set -ex
+
+GIT_REPO_PATH=https://github.com/btoll/go-algorand
+HASH=${1:-"rel/stable"}
+export HASH
+CHANNEL=${1:-"stable"}
+export CHANNEL
+export DEBIAN_FRONTEND=noninteractive
+
+sudo apt-get update -q
+sudo apt-get upgrade -q -y
+
+#umask 0077
+mkdir -p "${HOME}"/{.gnupg,go,gpgbin,dummyaptly,dummyrepo,prodrepo,tempkey}
+
+# Check out
+mkdir -p "${HOME}/go/src/github.com/algorand"
+cd "${HOME}/go/src/github.com/algorand" && git clone --single-branch --branch "${HASH}" "${GIT_REPO_PATH}" go-algorand
+# TODO: if we are checking out a release tag, `git tag --verify` it
+
+# Install latest Go
+# TODO: make a config file in root of repo with single source of truth for Go major-minor version
+cd "${HOME}"
+python3 "${HOME}/go/src/github.com/algorand/go-algorand/scripts/get_latest_go.py" --version-prefix=1.12
+# $HOME will be interpreted by the outer shell to create the string passed to sudo bash
+sudo bash -c "cd /usr/local && tar zxf ${HOME}/go*.tar.gz"
+
+GOPATH=${HOME}/go
+export GOPATH
+export PATH=${HOME}/gpgbin:${GOPATH}/bin:/usr/local/go/bin:${PATH}
+
+cat <<EOF>"${HOME}/gpgbin/remote_gpg_socket"
+export GOPATH=\${HOME}/go
+export PATH=\${HOME}/gpgbin:${GOPATH}/bin:/usr/local/go/bin:${PATH}
+gpgconf --list-dirs | grep agent-socket | awk -F: '{ print \$2 }'
+EOF
+
+chmod +x "${HOME}/gpgbin/remote_gpg_socket"
+
+sudo apt-get update
+sudo apt-get install -y build-essential automake autoconf awscli docker.io git gpg nfs-common python3 rpm sqlite3 python3-boto3 g++ libtool rng-tools
```

Comment:
again ? merge this with the upper portion

Reply:
Will do.